sdk/core/System.ClientModel/src/Message/PipelineRequestHeaders.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace System.ClientModel.Primitives;
+
+internal class PipelineRequestHeaders : PipelineMessageHeaders
+{
+    private ArrayBackedPropertyBag<IgnoreCaseString, object> _headers;
```

Comment:
ArrayBackedPropertyBag is optimized for two tuples and then it's actually more expensive for more than two tuples. I don't think there are many HTTP requests with just two headers, and so I worry that this decreases, not increases perf.

Reply:
I think when we originally benchmarked the difference between this and a hashtable, the breakeven point was 10+

Reply:
This is the way it's implemented in Azure.Core today.  I'll file an issue to follow up to see if we can improve perf for this case in both places.

Reply:
Also, @KrzysztofCwalina, at some point you had mentioned moving request to use the System.Net.Http HttpRequestMessage type when PipelineRequest is first created, instead of creating it in the transport right before sending.  We'd probably want to look at the perf implications of this, but let me know if you want me to file an issue to track that investigation since we didn't do it previously.

Reply:
Tracking with https://github.com/Azure/azure-sdk-for-net/issues/41229